tests/unit/detection/test_detection_model.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import pytest
+from torch import tensor
+from torchvision.models.detection.faster_rcnn import FasterRCNN
+
+from utils_cv.detection.dataset import DetectionDataset
+from utils_cv.detection.bbox import DetectionBbox
+from utils_cv.detection.model import (
+    get_pretrained_fasterrcnn,
+    DetectionLearner,
+    _get_det_bboxes,
+    _apply_threshold,
+    _calculate_ap,
+)
+
+
+@pytest.fixture(scope="session")
+def od_detection_dataset(tiny_od_data_path):
+    """ returns a basic detection dataset. """
+    return DetectionDataset(tiny_od_data_path)
+
+
+@pytest.fixture(scope="session")
+def od_detection_learner(od_detection_dataset):
+    """ returns a basic detection dataset. """
+    learner = DetectionLearner(od_detection_dataset)
+    learner.fit(1)
+    return learner
+
+
+def test__get_det_bboxes(od_sample_raw_preds, od_data_path_labels):
+    """ test that `_get_det_bboxes` can convert raw preds to DetectionBboxes. """
+    det_bboxes = _get_det_bboxes(
+        od_sample_raw_preds, labels=od_data_path_labels, im_path=None
+    )
+    assert type(det_bboxes[0]) == DetectionBbox
+    assert len(det_bboxes) == 5
+
+
+def test__apply_threshold(od_sample_detection_bboxes):
+    """ Test `_apply_threshold` and verify it works at different thresholds. """
+    det_bboxes = _apply_threshold(od_sample_detection_bboxes, threshold=0.5)
+    assert len(det_bboxes) == 3
+    det_bboxes = _apply_threshold(od_sample_detection_bboxes, threshold=0.01)
+    assert len(det_bboxes) == 5
+    det_bboxes = _apply_threshold(od_sample_detection_bboxes, threshold=0.995)
+    assert len(det_bboxes) == 2
+
+
+def test_get_pretrained_fasterrcnn():
+    """ Simply test that `get_pretrained_fasterrcnn` returns the correct type for now. """
+    assert type(get_pretrained_fasterrcnn(4)) == FasterRCNN
+
+
+def test_detection_learner_init(od_detection_dataset):
+    """ Tests detection learner basic init. """
+    learner = DetectionLearner(od_detection_dataset)
+    assert type(learner) == DetectionLearner
+
+
+def test_detection_learner_init_model(od_detection_dataset):
+    """ Tests detection learner with model settings. """
+    classes = len(od_detection_dataset.labels)
+    model = get_pretrained_fasterrcnn(num_classes=classes, min_size=600, max_size=2000)
+    learner = DetectionLearner(od_detection_dataset, model=model)
+    assert type(learner) == DetectionLearner
+    assert learner.model == model
+    assert learner.model != get_pretrained_fasterrcnn(classes)
+
+
+def test_detection_learner_train_one_epoch(od_detection_learner):
+    """ Simply test that a small training loop works. """
+    od_detection_learner.fit(epochs=1)
```

Comment:
[minor] I have a feeling this might be too slow for CPU unless we reduce the training set to only a couple of images, but we'll see.